### PR TITLE
Prepare v1.0.6 release automation and host cleanup

### LINKED ISSUE #187: Automate GitHub release creation from release tags
## Summary

The current release workflow publishes to npm when a `v*.*.*` tag is pushed, but it does not create the corresponding GitHub Release entry automatically. This was missed during the `v1.0.5` release and had to be fixed manually after the fact.

## Why this matters

A release should be complete from one tag push:

- npm package published
- GitHub Release entry created
- release notes attached consistently
- latest release status updated automatically

Right now the workflow only covers the npm side.

## Requested outcome

Extend the release automation so a tagged release also creates or updates the GitHub Release entry automatically.

## Release notes preference

Preserve the documented release-note preference from memory: blend curated/manual release notes with GitHub auto-generated release notes when possible, since auto-generation adds useful contributor/full-changelog tidbits even when the manual notes are better. Source: memory/2026-05-12.md#L12-L18

## Acceptance criteria

- pushing a release tag creates the GitHub Release entry automatically
- the workflow does not require a manual post-release `gh release create` step
- release notes support a curated/manual section plus GitHub-generated notes where practical
- reruns are safe and do not create duplicate releases
- the latest stable release is marked correctly

## Notes

The workflow already validates and publishes successfully on tag push; this issue is specifically about automating the GitHub Release object/page creation step.

### LINKED ISSUE #188: Move package.json shell one-liners into readable script files
## Summary

`package.json` currently contains several hardcoded inline scripts and shell one-liners (for example `build` and `check:version-sync`). They work, but they are harder to read, test, review, and evolve than dedicated script files under a `scripts/` directory.

## Why this matters

Inline command strings are fine for tiny glue, but once they encode real logic they become:

- harder to understand in review
- harder to test directly
- harder to reuse from CI or other scripts
- easier to break with quoting or platform-specific shell behavior

## Requested outcome

Move non-trivial logic out of `package.json` into readable, dedicated script files under `scripts/` (or a similarly explicit scripts directory), while keeping `package.json` as a thin command map.

## Scope examples

At minimum, review hardcoded logic such as:

- `build`
- `check:version-sync`

And evaluate whether other non-trivial inline commands should move too.

## Acceptance criteria

- non-trivial inline logic is moved into dedicated script files
- script files are named clearly and remain cross-platform
- `package.json` commands stay simple wrappers around those files
- the extracted scripts are easy to test in isolation where appropriate
- validation/build/release behavior stays unchanged after the refactor

## Non-goal

This is not a request to change release semantics or validation policy; it is a readability/testability refactor.

### LINKED ISSUE #190: Audit and fix remaining stale multi-intent wording in playbooks
## Summary

A focused audit for intent-related wording after the `1.0.5` docs cleanup found a couple of remaining stale prompt placeholders that still describe the input as a single/primary intent even though the current implementation supports additive repeated `--intent` values.

This is a follow-up to the earlier docs cleanup in #185, which fixed README, setup, and breadth guidance but did not fully catch every playbook prompt block.

### 1. `RECOMMENDATION-POLICY-PLAYBOOK.md`

Current text:

```text
Intent: <optional primary intent>
```

Why this is stale:

- recommendation flows now support repeated additive `--intent` values
- only the first intent remains primary for backward-compatible output/context
- describing the whole input as only an "optional primary intent" is misleading

Relevant implementation/docs evidence:

- `src/recommend/commands.ts` documents `--intent` as repeatable with additive merging
- `src/recommend/report.ts` records only the first intent as primary for backward compatibility
- `README.md` and other updated playbooks already describe repeated/additive intents

### 2. `AI-ENRICHMENT-PLAYBOOK.md`

Current text:

```text
Intent: <optional primary intent>
```

Why this is stale:

- the playbook guides `workspace` / recommendation-adjacent flows that now accept repeated additive intents
- the same compatibility nuance applies here: first intent can remain primary, but the input itself is no longer single-intent-only

Relevant implementation/docs evidence:

- `src/workspace.ts` documents `--intent` as repeatable with additive merging
- `README.md` and updated playbooks already describe repeated/additive intents

## Audit notes

Repo-wide search was specifically aimed at similar stale wording such as:

- `optional primary intent`
- `single-value option`
- `Only one --intent value is allowed`
- `Pick one primary intent per run`

After validating the hits against current code, these two prompt placeholders were the remaining verified stale wording. Intentional references to the first intent remaining primary for backward compatibility should stay.

## Acceptance criteria

- update the remaining stale prompt placeholders in:
  - `RECOMMENDATION-POLICY-PLAYBOOK.md`
  - `AI-ENRICHMENT-PLAYBOOK.md`
- preserve the accurate nuance that the first provided intent may still remain primary where backward-compatible output/context needs it
- avoid reintroducing wording that implies repeated `--intent` flags are unsupported

## Non-goal

Do not remove intentional code/docs references that describe the first intent as primary for activation/report compatibility. This issue is only about stale wording that misdescribes the user input shape itself.

### LINKED ISSUE #191: Standardize VS Code to one canonical host name across user-facing command surfaces
## Summary

A newly observed bug showed that `agent-harness recommend policy:print --host vscode` fails because the recommendation command currently requires the internal canonical host id `copilot-vscode` instead of the user-facing VS Code host name.

The successful workaround was:

```bash
agent-harness recommend policy:print --host copilot-vscode
```

## Why this matters

The CLI already presents VS Code as `vscode` across major user-facing flows such as:

- `workspace vscode`
- `wire vscode`
- `install native --host vscode`
- `setup doctor --host vscode`

Requiring `copilot-vscode` in some recommendation/policy commands leaks an internal host id into the public CLI surface and creates an inconsistent mental model.

## Requested outcome

Audit and update all relevant user-facing command surfaces so VS Code uses one canonical host name consistently.

## Suggested direction

Use `vscode` as the user-facing canonical name and map it internally to `copilot-vscode` where the internal lifecycle/recommendation ids still need to exist.

## Scope

At minimum, audit and update:

- `recommend policy:print --host ...`
- `recommend explain --host ...`
- `recommend ai-review --host ...`
- related help text, docs, and playbook examples that currently expose `copilot-vscode` as the command argument for VS Code

## Acceptance criteria

- user-facing recommend/policy command surfaces accept `--host vscode`
- help text and docs consistently show the chosen VS Code canonical host name for those command surfaces
- internal host ids can remain internal where necessary, but should not leak into normal command usage without a clear reason
- validation passes after the change

## Non-goal

This is not a request to rename internal policy file paths or lifecycle ids if keeping `copilot-vscode` internally remains the cleaner implementation detail.

### PR DESCRIPTION
**Pull request by ar27111994**
## Summary
- automate GitHub Release sync for version tags using curated changelog notes plus GitHub-generated notes
- move non-trivial build, version-sync, and release-note logic out of package.json into readable scripts with direct script-level tests
- standardize user-facing recommend/policy VS Code host naming on `vscode` while preserving the internal `copilot-vscode` recommendation host
- fix the remaining stale multi-intent wording in the recommendation and AI-enrichment playbooks
- prepare the v1.0.6 changelog and package metadata

## Validation
- npm run validate
- npm run validate:release

## Issue coverage
Closes #187
Closes #188
Closes #190
Closes #191

#189 is a duplicate of #190 and is covered by the same implementation.
#192 is intentionally deferred for a later opt-in change; v1.0.6 keeps env override behavior limited to `recommendationLimit` without automatic scaling.

## Summary by CodeRabbit

* **New Features**
  * Auto-create/update GitHub Release pages from v* tags, merging changelog notes and safe to re-run; added a release job triggered on version tags.
  * Improved CLI host handling so host names/aliases display consistently (e.g., vscode).

* **Documentation**
  * Normalized examples to use --host vscode and clarified multi-intent wording in playbooks and README.

* **Chores**
  * Bumped package to 1.0.6 and moved build/release/version checks into reusable scripts; added a CLI command to sync GitHub releases.

* **Tests**
  * Added regression tests for version-sync, release-notes extraction/merge, release sync payloads, and host normalization.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ar27111994/agent-harness/pull/193)

<h3>Greptile Summary</h3>

This PR prepares the v1.0.6 release by automating GitHub Release sync on version tags, extracting build/version-check/release-note logic from package.json inline scripts into dedicated `scripts/` modules with regression tests, and standardizing the user-facing VS Code host name to `vscode` while preserving the internal `copilot-vscode` mapping.

- **Release automation**: a new `github-release` workflow job calls `scripts/sync-github-release.mjs` after publish; the script fetches GitHub-generated notes, merges them with curated CHANGELOG entries, and creates or updates the GitHub Release page in a rerun-safe way.
- **Host normalization**: `resolveRecommendationHost` and `formatRecommendationHostForDisplay` in `src/recommend/hosts.ts` provide a clean alias layer so users see `vscode` in CLI help and error messages while the internal `copilot-vscode` host identifier is unchanged; backward-compat input of `copilot-vscode` still resolves correctly.
- **Documentation**: `README.md`, `AI-ENRICHMENT-PLAYBOOK.md`, and `RECOMMENDATION-POLICY-PLAYBOOK.md` updated to use `--host vscode` in all user-facing examples; multi-intent prompt wording fixed in both playbooks.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the changes are well-scoped, all new code paths are covered by unit tests, and both issues flagged in the prior review round have been addressed.

The release-sync script correctly resolves paths through process.cwd(), guards execution behind isDirectExecution, and handles the create/update distinction cleanly. The host-alias layer in hosts.ts preserves backward compatibility while presenting the correct user-facing name. No live side effects are introduced in importable modules, and the new workflow job carries the minimum required permissions.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| .github/workflows/release.yml | Adds a github-release job after publish; uses pinned SHA actions, sets required contents:write permission, and correctly gates on v* tag refs. |
| scripts/sync-github-release.mjs | New GitHub Release sync script. Both previous thread issues are resolved: isDirectExecution guard is present and package.json is read via resolve(process.cwd(), 'package.json'). |
| scripts/release-notes.mjs | Extracts CHANGELOG sections and merges with GitHub-generated notes. VERSION_HEADING_PATTERN lacks the g flag so .exec() is stateless. |
| scripts/build.mjs | Extracts build logic from package.json inline script into a readable module with isDirectExecution guard. |
| scripts/check-version-sync.mjs | Validates package.json/package-lock.json version triple with distinct error messages. Has isDirectExecution guard. |
| src/recommend/hosts.ts | Adds resolveRecommendationHost, formatRecommendationHostForDisplay, and getRecommendationHostChoices with correct vscode->copilot-vscode mapping. |
| src/recommend/commands.ts | Replaces isRecommendationHost checks with resolveRecommendationHost throughout; error messages now show user-facing host choices. |
| scripts/tests/sync-github-release.test.mjs | Tests buildReleasePayload for stable/prerelease create/update combinations and getOptionValue edge cases. |
| scripts/tests/release-notes.test.mjs | Covers extraction, missing-version error, all four note-combination cases, and tag normalisation helpers. |
| src/tests/recommend-hosts.test.ts | Tests vscode/copilot-vscode normalisation helpers and an integration-style policy:print invocation. |

</details>

<sub>Reviews (6): Last reviewed commit: ["Set GitHub release user agent"](https://github.com/ar27111994/agent-harness/commit/6c5b2a291ee01f9f88f1e8f0589115b7aa6021c5) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31970012)</sub>

### CODE REVIEW

File: scripts/release-notes.mjs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  if (!headingMatch) {
+    throw new Error(`Could not find CHANGELOG entry for version ${version}.`);
+  }
+
+  const sectionStart = headingMatch.index + headingMatch[0].length;
+  const remainder = changelogText.slice(sectionStart);
+  VERSION_HEADING_PATTERN.lastIndex = 0;
+  const nextHeadingMatch = VERSION_HEADING_PATTERN.exec(remainder);
+  const sectionEnd =
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply scripts/release-notes.mjs]: Already addressed in f5f189b: the release-notes helper no longer uses a shared global-state regex (g flag and lastIndex mutation were removed), so this thread is obsolete.

File: scripts/sync-github-release.mjs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+function resolveReleaseContext() {
+  const repo =
+    getOptionValue("--repo") ??
+    getRequiredEnvironmentValue("GITHUB_REPOSITORY");
+  const tag =
+    getOptionValue("--tag") ?? getRequiredEnvironmentValue("GITHUB_REF_NAME");
+  const token =
+    process.env.GITHUB_TOKEN ??
+    process.env.GH_TOKEN ??
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply scripts/sync-github-release.mjs]: Fixed in 4789d90: token selection now treats empty strings as missing instead of attempting authenticated requests with an empty token.

File: scripts/sync-github-release.mjs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+function buildReleasePayload(context, combinedNotes) {
+  return {
+    tag_name: context.tag,
+    target_commitish: context.targetCommitish,
+    name: context.tag,
+    body: combinedNotes,
+    draft: false,
+    prerelease: isPreRelease(context.version),
+    make_latest: isPreRelease(context.version) ? "false" : "true",
+  };
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply scripts/sync-github-release.mjs]: Fixed in 4f66a57: update payloads now omit make_latest, and there is a regression test covering create-vs-update payload behavior so reruns cannot flip latest inadvertently.

File: src/recommend/hosts.ts [hunk 4]
```diff
@@ -27,3 +31,51 @@ export function isRecommendationHost(
...
+    if (
+      adapter.id === value ||
+      adapter.aliases.includes(value) ||
+      formatRecommendationHostForDisplay(adapter.recommendationHost) === value
+    ) {
+      return adapter.recommendationHost;
+    }
+  }
+
+  return value ===
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply src/recommend/hosts.ts]: Fixed in 4f66a57: 
esolveRecommendationHost() now trims and lowercases CLI input before matching, and the host test suite covers mixed-case/whitespace input.

File: scripts/sync-github-release.mjs [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+const GITHUB_API_URL = "https://api.github.com";
+const API_VERSION = "2022-11-28";
+
+function getOptionValue(flagName) {
+  const index = process.argv.indexOf(flagName);
+  if (index === -1) {
+    return undefined;
+  }
+
+  return process.argv[index + 1];
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply scripts/sync-github-release.mjs]: Fixed in 4f66a57: getOptionValue() now rejects missing or flag-like values with a clear error, and script tests cover both failure cases.

File: scripts/sync-github-release.mjs [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+  const response = await fetch(`${GITHUB_API_URL}${endpoint}`, {
+    method,
+    headers: {
+      Accept: "application/vnd.github+json",
+      Authorization: `Bearer ${token}`,
+      "Content-Type": "application/json",
+      "X-GitHub-Api-Version": API_VERSION,
+    },
+    body: body === undefined ? undefined : JSON.stringify(body),
+  });
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply scripts/sync-github-release.mjs]: Fixed in 64c0206: githubRequest() now uses a default 30s AbortSignal timeout, supports optional timeoutMs overrides, and rethrows timeout/abort failures with a clear error message. The prerelease payload test follow-up is also covered in the same commit.

File: scripts/sync-github-release.mjs [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  const timeoutSignal = AbortSignal.timeout(timeoutMs);
+
+  let response;
+  try {
+    response = await fetch(`${GITHUB_API_URL}${endpoint}`, {
+      method,
+      headers: {
+        Accept: "application/vnd.github+json",
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
```
[the remark this answers is not in the document]
Comment (ar27111994) [reply scripts/sync-github-release.mjs:87]: Fixed in 6c5b2a2: the release script now sends a stable \User-Agent: agent-harness\ header on GitHub API requests, matching the repo’s other GitHub fetch path. Validated with \
pm run validate\ and \
pm run validate:release\.
